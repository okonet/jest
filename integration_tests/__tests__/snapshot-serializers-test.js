/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */
'use strict';

const path = require('path');
const skipOnWindows = require('skipOnWindows');
const {cleanup} = require('../utils');
const runJest = require('../runJest');

const testDir = path.resolve(__dirname, '../snapshot-serializers');
const snapshotsDir = path.resolve(testDir, '__tests__/__snapshots__');
const snapshotPath = path.resolve(snapshotsDir, 'snapshot-test.js.snap');

const runAndAssert = () => {
  const result = runJest.json('snapshot-serializers');
  const json = result.json;
  expect(json.numTotalTests).toBe(3);
  expect(json.numPassedTests).toBe(3);
  expect(json.numFailedTests).toBe(0);
  expect(json.numPendingTests).toBe(0);
  expect(result.status).toBe(0);
};

describe('Snapshot serializers', () => {
  skipOnWindows.suite();

  beforeEach(() => cleanup(snapshotsDir));
  afterEach(() => cleanup(snapshotsDir));

  it('renders snapshot', () => {
    runAndAssert();
    const snapshot = require(snapshotPath);
    expect(snapshot).toMatchSnapshot();
  });

  it('compares snapshots correctly', () => {
    // run twice, second run compares result with snapshot from first run
    runAndAssert();
    runAndAssert();
  });
});
