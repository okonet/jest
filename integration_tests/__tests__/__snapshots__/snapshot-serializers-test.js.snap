exports[`Snapshot serializers renders snapshot 1`] = `
Object {
  "snapshot serializers works with default serializers 1": "
<div
  id="foo" />
",
  "snapshot serializers works with first plugin 1": "foo: 1",
  "snapshot serializers works with nested serializable objects 1": "foo: bar: 2",
  "snapshot serializers works with prepended plugins and default serializers 1": "
<div
  aProp={
    Object {
      "a": 6,
    }
  }
  bProp={foo: 8} />
",
  "snapshot serializers works with second plugin 1": "bar: 2",
}
`;
